samplePDF/samplePDFFDBase.cpp

```diff
@@ -100,105 +253,77 @@ void samplePDFFDBase::SetupSampleBinning(){
   dathist2d = new TH2D("d"+histname2d+samplename,histtitle, 1, 0, 1, 1, 0, 1);
 
   //Make some arrays so we can initialise _hPDF1D and _hPDF2D with these
-  XBinEdges.reserve(SampleNXBins);
-  YBinEdges.reserve(SampleNYBins);
-  std::cout << "XBinning: " << std::endl;
-  for(unsigned XBin_i = 0 ; XBin_i < SampleNXBins ; XBin_i++){
-	XBinEdges.push_back(SampleXBins[XBin_i]);
-	std::cout << SampleXBins[XBin_i] << ", ";
-  }
-  std::cout << "\n" << std::endl;
+  XBinEdges.reserve(SampleXBins.size());
+  YBinEdges.reserve(SampleYBins.size());
 
-  //And now the YBin Edges
-  std::cout << "YBinning: " << std::endl;
-  for(unsigned YBin_i = 0 ; YBin_i < SampleNYBins ; YBin_i++){
-	YBinEdges.push_back(SampleYBins[YBin_i]);
-	std::cout << SampleYBins[YBin_i] << ", ";
+  //A string to store the binning for a nice print out
+  std::string XBinEdgesStr = "";
+  std::string YBinEdgesStr = "";
+
+  for(auto XBinEdge : SampleXBins){
```

Comment:
I think this is unnecessary. spdlog doesn't seem to include all of the fmtlib features, but we can use its bundled fmtlib.

```c++
//use spdlog bundled fmtlib ranges
#include "spdlog/fmt/bundled/ranges.h"

//...

MACH3LOG_INFO(fmt::format("XBinning: {}",SampleXBins));
```